lib/cereal/builders/entity.ex

```diff
@@ -21,6 +24,18 @@ defmodule Cereal.Builders.Entity do
     }
   end
 
+  # if an assigns/2 function was specified in the serializer - we'll add its
+  # return value to the conn's assigns
+  defp do_assigns(%{serializer: serializer, conn: conn} = context) do
+    assigns =
+      Map.get(conn, :assigns, %{})
+      |> Map.merge(serializer.assigns(context.data, conn))
+    conn =
+      context.conn
+      |> Map.put(:assigns, assigns)
```

Comment:
Since `conn.assigns` is a data structure managed by Plug.Conn and it's recommended to update it via `Plug.Conn.assign/3`, I wonder if we should do that (instead of merging manually)? Thinking like:

```elixir
import Plug.Conn, only: [assign: 3]

conn =
  context.data
  |> serializer.assigns(conn)
  |> Enum.reduce(conn, fn {key, value}, acc -> assign(acc, key, value) end)

Map.put(context, :conn, conn)
```